### ISSUE TITLE
Login via omniauth doesn't work correctly for system tests.

### ISSUE THREAD

**Issue by BLauris** [omniauth/omniauth#908]
- Provider Gem: `omniauth-facebook`
- Ruby Version: `2.4.1`
- Framework: `Rails 5.1.2`

On the project, I am working on there are several integration tests.
`Login` is set up like in the wiki and everything was working just fine.

A few days ago I had to move to `system tests`. 
Right now my tests are failing because of `omniauth` logging in only one user.
For example.
I have two tests and each of them logs in different user and what happens is that for the second test 
it logs in again the first user, not the second one.

So, basically what happens is, it will continue to log in the user that was logged in on the first test to all tests.

**Comment by wnm** [comment]
I'm experiencing the same issue.

I'v set up the tests like this:

```ruby
setup do
  shop = shops(:first)
  
  OmniAuth.config.add_mock(:shopify,
    {
      provider: "shopify",
      uid: shop.shopify_domain,
      credentials: {
        token: shop.shopify_token
      }
    }
  )

  visit "/login?shop=#{shop.shopify_domain}"
end

test "some test" do
 ...
end
```

In a different test, I'm using a different shop fixture, but it logs in the one from the first test.

Somehow, [mock_auth_for](https://github.com/omniauth/omniauth/blob/master/lib/omniauth.rb#L130) always returns the first one....

**Comment by wnm** [comment]
After more debugging I'm convinced this is because right now puma is run in clustered mode for system tests. This will be [fixed](https://github.com/rails/rails/pull/30638) in rails 5.2.0
